Add Pass & Drop filters, Refactor `Match` config

XAMPPRocky opened the pull request:
This adds simple "pass" and "drop" filters, that either always allow a packet to pass, or always drop a packet. These filters aren't useful on their own, but become useful when used in combination with conditional flow filters like `Match` (#447).

closes #454

markmandel commented:
Only question I have: Did we want to include the documentation with these Filters in this PR, or should it come as a separate? (if separate, let's track that in an issue, so we don't forget).

Otherwise, LGTM!

XAMPPRocky commented:
I've added documentation, and I've also refactored a small bit of `Matches`, I've now renamed it to `Match`, and I've updated it from having special cased `Pass` and `Drop` branches in its configuration, if you want to drop or pass a packet, you should use the `Pass` and `Drop` filters. This simplifies some of the configuration.

@markmandel Would nice to get this in the next release, so that it's not technically a breaking change.

markmandel commented on src/filters.rs:
```diff
@@ -30,11 +30,13 @@ pub mod capture;
 pub mod load_balancer;
 pub mod local_rate_limit;
-pub mod matches;
+pub mod r#match;
```
_(non blocking):_ I assume the `r#` is because `match` is a keyword?
